### PR TITLE
feat: Add Ruby Language Support

### PR DESCRIPTION
**Pull request by CrCs2O4**
Thanks a lot for a great project! Happy to contribute support of the Ruby programming language (based on [Support for golang](https://github.com/TabbyML/tabby/issues/553))
- [x] Stop words support
- [x] Repository context support
- [x] Update documentation / changelog

also
- closes [Ruby Support#89](https://github.com/TabbyML/tabby/issues/89)

### CODE REVIEW

File: crates/tabby-scheduler/queries/ruby.scm [hunk 1]
```diff
@@ -0,0 +1,66 @@
+; https://github.com/tree-sitter/tree-sitter-ruby/blob/master/queries/tags.scm
```
Comment (wsxiaoys) [inline crates/tabby-scheduler/queries/ruby.scm:1]: If we're only using the default tags, you can refer to the exported variable `TAG_QUERY` from tree-sitter-ruby directly, just like the Rust or Python configurations.

File: website/docs/programming-languages.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ For an actual example of an issue or pull request adding the above support, plea
 * [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript)
 * [TypeScript](https://www.typescriptlang.org/)
 * [Golang](https://go.dev/): Since v0.4.0
+* [Ruby](https://www.ruby-lang.org/)
```
Comment (wsxiaoys) [inline website/docs/programming-languages.md:29]:
```suggestion
* [Ruby](https://www.ruby-lang.org/): Since v0.4.0
```

**Review by wsxiaoys — CHANGES_REQUESTED** [review]
Thanks for the PR! Adding a few comments

File: crates/tabby-common/assets/languages.toml [hunk 3]
```diff
@@ -50,4 +50,52 @@ top_level_keywords = [
...
     "const",
-]
\ No newline at end of file
+]
+
+[[config]]
+languages = ["ruby"]
+line_comment = "#"
+# from https://docs.ruby-lang.org/en/3.2/keywords_rdoc.html
+top_level_keywords = [
```
[the remark this answers is not in the document]
Comment (CrCs2O4) [reply crates/tabby-common/assets/languages.toml:58]: you're right, did a cleanup 👍

File: crates/tabby-scheduler/queries/ruby.scm (hunk 1 above)
Comment (CrCs2O4) [reply crates/tabby-scheduler/queries/ruby.scm]: updated 👍

File: crates/tabby-scheduler/src/dataset.rs [hunk 4]
```diff
@@ -270,6 +270,17 @@ lazy_static! {
                     .unwrap(),
                 ),
             ),
+            (
+                "ruby",
+                TagsConfigurationSync(
+                    TagsConfiguration::new(
+                        tree_sitter_ruby::language(),
+                        tree_sitter_rust::TAGGING_QUERY,
```
Comment (KenLau) [inline crates/tabby-scheduler/src/dataset.rs:278]: should this line be tree_sitter_ruby::TAGGING_QUERY?

Comment (wsxiaoys) [reply crates/tabby-scheduler/src/dataset.rs:278]: this has been fixed in main

**Pull request merged**